### ISSUE TITLE
GKE Private Cluster Subnetwork Error

### ISSUE THREAD

**Issue by soudaburger** [rancher/rancher#22693]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Create GKE Cluster, paste JSON key
2. Node Subnet is dropdown (no ability to NOT select anything)
3. Use IP Aliases -> YES
4. Create Subnetwork -> Enabled, cannot choose "Disabled".
5. Private Cluster -> Enabled

**Result:**
Googleapi: Error 400: Create_subnetwork cannot be used with an existing subnetwork., badRequest

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.2.8
- Installation option (single install/HA): HA
Installed on GKE

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): GKE
- Machine type (cloud/VM/metal) and specifications (CPU/memory): n1-standard-2
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-20T04:49:16Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.8-gke.10", GitCommit:"f53039cc1e5295eed20969a4f10fb6ad99461e37", GitTreeState:"clean", BuildDate:"2019-06-19T20:48:40Z", GoVersion:"go1.10.8b4", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
(paste the output here)
```

So essentially, it boils down to the forced drop down of selecting a subnetwork, then requesting to deploy a cluster using IP aliases.  This then forces the "Create Subnetwork" option which conflicts with the subnetwork dropdown.

There should be an option do auto-create or otherwise ignore the subnetwork dropdown and/or select "disabled" for create subnetwork.

**Comment by soudaburger** [comment]
![Screen Shot 2019-09-06 at 2 05 37 PM](https://user-images.githubusercontent.com/29105445/64454449-9b4f5580-d0b0-11e9-8c42-92cd638ffcc3.png)
![Screen Shot 2019-09-06 at 2 07 05 PM](https://user-images.githubusercontent.com/29105445/64454450-9b4f5580-d0b0-11e9-83ef-4f2214f628c4.png)

**Comment by soudaburger** [comment]
This effectively means I cannot create a private cluster.  You cannot select private cluster unless you enable IP aliases.  But you can't select IP aliases without force-enabling "Create Subnetwork", so the provisioning ultimately fails.

**Comment by gbates101** [comment]
Running into the same problem here. I want to use Rancher create a private cluster (read, no public IPs) on an existing subnet, and should not need to enable IP aliasing to do so.

**Comment by gbates101** [comment]
After a little digging and testing, it seems that a [private cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters) in GKE cannot be created without using IP Alias and a dedicated subnetwork, hence the form restrictions. I even edited the HTML for the GKE cluster creation in Rancher to force every permutation of settings, and I get Googleapi errors if I do so. Regardless, what I was ultimately interested in is a setting to use internal IPs only, and everything else be the same.

tl;dr: I didn't know what I was doing, now I do, and I believe this form is working appropriately.

**Comment by flaccid** [comment]
Was not able to replicate the existing error with rancher 2.5.3. Its just about mapping the form options to https://cloud.google.com/kubernetes-engine/docs/concepts/private-cluster-concept#overview for example, the last option of full public you do need to select a vpc in addition to disabled the master authorized networks.
